sdk/tables/azure-data-tables/azure/data/tables/aio/_table_client_async.py

```diff
@@ -415,19 +415,18 @@ async def update_entity(
         """
         match_condition = kwargs.pop("match_condition", None)
         etag = kwargs.pop("etag", None)
-        if match_condition and entity and not etag:
+        if match_condition and not etag:
             try:
-                etag = entity.metadata.get("etag", None)  # type: ignore
+                etag = entity.metadata.get("etag", None)  # type: ignore[union-attr]
             except (AttributeError, TypeError):
                 pass
-        if_match = _get_match_headers(
-            etag=etag,
-            match_condition=match_condition or MatchConditions.Unconditionally,
+        match_condition = _get_match_condition(
+            etag=etag, match_condition=match_condition or MatchConditions.Unconditionally
         )
-
+        entity = _add_entity_properties(entity)
         partition_key = entity["PartitionKey"]
         row_key = entity["RowKey"]
-        entity = _add_entity_properties(entity)
```

Comment:
This one makes me a bit nervous.... because technically the async client was doing this the right way, and the sync client was doing this incorrectly. I didn't realize we had differences between the clients here.
The `upsert` method is also doing this the other way around. Could we revert this for now and open an issue to explore this as part of the encoder work (or add it to the existing encoder issue)?